addons/binding/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/handler/CoffeeSystemChannelSelector.java

```diff
@@ -0,0 +1,144 @@
+
+    PRODUCT_TYPE("productTypeId", "productType", StringType.class, true),
+    DEVICE_TYPE("mieleDeviceType", "deviceType", StringType.class, true),
+    BRAND_ID("brandId", "brandId", StringType.class, true),
+    COMPANY_ID("companyId", "companyId", StringType.class, true),
+    STATE("state", "state", StringType.class, false),
+    PROGRAMID("programId", "program", StringType.class, false),
+    PROGRAMTYPE("programType", "type", StringType.class, false),
+    PROGRAMPHASE("phase", "phase", StringType.class, false),
+    // lightingStatus signalFailure signalInfo
+    DOOR("signalDoor", "door", OpenClosedType.class, false) {
+        @Override
+
+        public State getState(String s, DeviceMetaData dmd) {
+            if (s.equals("true")) {
+                return getState("OPEN");
+            }
+
+            if (s.equals("false")) {
+                return getState("CLOSED");
+            }
+
+            return UnDefType.UNDEF;
+        }
+    },
+    SWITCH(null, "switch", OnOffType.class, false);
+
+    private final Logger logger = LoggerFactory.getLogger(CoffeeSystemChannelSelector.class);
+
+    private final String mieleID;
+    private final String channelID;
+    private final Class<? extends Type> typeClass;
+    private final boolean isProperty;
+
+    private CoffeeSystemChannelSelector(String propertyID, String channelID, Class<? extends Type> typeClass,
+            boolean isProperty) {
+        this.mieleID = propertyID;
+        this.channelID = channelID;
+        this.typeClass = typeClass;
+        this.isProperty = isProperty;
+    }
+
+    @Override
+    public String toString() {
+        return mieleID;
+    }
+
+    @Override
+    public String getMieleID() {
+        return mieleID;
+    }
+
+    @Override
+    public String getChannelID() {
+        return channelID;
+    }
+
+    @Override
+    public Class<? extends Type> getTypeClass() {
+        return typeClass;
+    }
+
+    @Override
+    public boolean isProperty() {
+        return isProperty;
+    }
+
+    @Override
+    public State getState(String s, DeviceMetaData dmd) {
+        if (dmd != null) {
+            String localizedValue = getMieleEnum(s, dmd);
+            if (localizedValue == null) {
+                localizedValue = dmd.LocalizedValue;
+            }
+            if (localizedValue == null) {
+                localizedValue = s;
+            }
+
+            return getState(localizedValue);
+        } else {
+            return getState(s);
+        }
+    }
+
+    public State getState(String s) {
+        try {
+            Method valueOf = typeClass.getMethod("valueOf", String.class);
+            State state = (State) valueOf.invoke(typeClass, s);
+            if (state != null) {
+                return state;
+            }
+        } catch (Exception e) {
+            logger.error("An exception occurred while converting '{}' into a State", s);
+        }
+
+        return null;
```

Comment:
Since various other classes are implemented exactly the same way, I would assume, that `null` is correct here. `UnDefType.UNDEF` is used in other places, so I would think, that `null` is intended. Besides that, `org.eclipse.smarthome.core.type.State` also returns `null` in one case - which is the return type as far as I understand.

Reply:
It's probably correct to return null. I was triggered because in the enum override getState in DOOR above does return a `UnDefType.UNDEF` in some case so it seems logically to do the same here. Also recent development with smart home regarding null values (for example annotating a class no null by default and specifying what does handle/return null values. When null can be avoided for such cases it makes development easier)